src/App.tsx

```diff
@@ -1,61 +1,156 @@
 import './App.scss';
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUserById(userId: number) {
+  const foundUser = usersFromServer.find((user: User) => (
+    user.id === userId
+  ));
+
+  return foundUser || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map((todo) => {
+  return {
+    ...todo,
+    user: getUserById(todo.userId),
+  };
+});
+
+const getMaxId = (todos: Todo[]) => {
+  return todos.reduce((acumulator: number, todo: Todo) => {
+    if (todo.id > acumulator) {
+      return todo.id;
+    }
+
+    return acumulator;
+  }, 0);
+};
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [todos, setTodos] = useState<Todo[]>(preparedTodos);
+  const [isTitle, setIsTitleError] = useState(false);
+  const [isUser, setIsUser] = useState(false);
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setIsTitleError(false);
+    setTitle(value);
+  };
+
+  const handleChangeSelect = (event: ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setIsUser(false);
+    setUserId(Number(value));
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    const titleWithoutExtraSpaces = title.trim();
+
+    if (!userId) {
+      setIsUser(true);
+    }
+
+    if (!titleWithoutExtraSpaces) {
+      setIsTitleError(true);
+    }
+
+    if (!titleWithoutExtraSpaces || !userId) {
+      return;
+    }
+
+    const id = getMaxId(todos) + 1;
+    const user = getUserById(Number(userId));
+
+    setTodos((currentTodos) => {
+      return [...currentTodos, {
+        id,
+        title: titleWithoutExtraSpaces,
+        completed: false,
+        userId,
+        user,
+      }];
+    });
+
+    if (title && userId) {
+      setTitle('');
+      setUserId(0);
+    }
```

Comment:
you already check it
```suggestion
    setTitle('');
    setUserId(0);
```